clang/lib/DPCT/DPCT.cpp

```diff
@@ -826,6 +826,23 @@ int runDPCT(int argc, const char **argv) {
 
     Tool.mapVirtualFile(SourcePathList[0], SourceCode);
 
+    static const std::string UnsupportedStr{"// UNSUPPORTED:"};
+    bool win_unsupported = false;
+    bool lin_unsupported = false;
+    if (SourceCode.starts_with(UnsupportedStr)) {
```

Comment:
The `cudaGraphicsD3D11RegisterResource` API is only migrated on Windows. 
On Linux, dpct returns below output:

`dpct exited with code: 1 (Migration not necessary; no CUDA code detected.)`

This is because this API is present in the header `cuda_d3d11_interop.h` which is only available on Windows platform.

Hence, I've devised this logic to parse supported and unsupported plaatforms.
Please let me know your thoughts on the same.

Reply:
ok, currently I have no better solution, just keep this solution is ok to me.